src/components/Container.js

```diff
@@ -131,58 +120,72 @@ class Container extends React.Component{
   }
 
   indexNamebyElevation = () => {
-    const { selectedTrail } = this.state;
-    let newHydrants = _.cloneDeep(this.state.hydrants);
+    const { selectedTrail, hydrants } = this.state;
 
-    const sortedTrailHydrants = _.chain(newHydrants)
-      .pickBy((h) => h['trail'] === selectedTrail)
+    const sortedTrailHydrants = hydrants.toJS()
+      .filter((h) => h.trail === selectedTrail)
       .orderBy('elevation', 'desc')
       .map((h,i)=> {
         h.name = i + 1
         return h
       }).value();
 
-    _.each(sortedTrailHydrants , (h,key)=> {
-      newHydrants[h.id] = h
+    let newHydrants = hydrants.map((h) => {
+      const elevationIndex = _.findIndex(sortedTrailHydrants, (sortedHydrant) => sortedHydrant.id === h.get('id'));
+      return h.set('name', elevationIndex);
     });
+
     this.setState({
       hydrants: newHydrants
     });
   }
 
-  mapControlClicked = (type) => {
-    if (type === 'Hydrant') {
-      this.setState({createType: type});
-    }else {
-      this.setState({createType: type || null, selectedTrail: null});
+  importKMLClicked = (klmData) => {
+    console.log(klmData)
+    if( this.state.mode === 'trails'){
```

Comment:
Is this necessary to distinguish between trail and hydrant KML's? 

If trails are always series of coordinates and hydrants are always points, why can't the upload function just distinguish which is which and put them in state without the user needing to be in a certain menu mode? That could allow imports of mixed trails and hydrants in the future as well without any changes.